pygments/lexers/lisp.py

```diff
@@ -2846,3 +2846,351 @@ class FennelLexer(RegexLexer):
+        'pp', 'prewalk', 'prin', 'prinf', 'print', 'printf',
+        'product', 'propagate', 'put', 'put-in', 'quit',
+        'range', 'reduce', 'reduce2', 'repl', 'require',
+        'resume', 'return', 'reverse', 'reverse!',
+        'run-context', 'sandbox', 'scan-number', 'setdyn',
+        'signal', 'slice', 'slurp', 'some', 'sort', 'sort-by',
+        'sorted', 'sorted-by', 'spit', 'string',
+        'string/ascii-lower', 'string/ascii-upper',
+        'string/bytes', 'string/check-set', 'string/find',
+        'string/find-all', 'string/format', 'string/from-bytes',
+        'string/has-prefix?', 'string/has-suffix?',
+        'string/join', 'string/repeat', 'string/replace',
+        'string/replace-all', 'string/reverse', 'string/slice',
+        'string/split', 'string/trim', 'string/triml',
+        'string/trimr', 'string?', 'struct', 'struct/getproto',
+        'struct/proto-flatten', 'struct/to-table',
+        'struct/with-proto', 'struct?', 'sum', 'symbol',
+        'symbol/slice', 'symbol?', 'table', 'table/clear',
+        'table/clone', 'table/getproto', 'table/new',
+        'table/proto-flatten', 'table/rawget', 'table/setproto',
+        'table/to-struct', 'table/weak', 'table/weak-keys',
+        'table/weak-values', 'table?', 'take', 'take-until',
+        'take-while', 'thaw', 'trace', 'true?', 'truthy?',
+        'tuple', 'tuple/brackets', 'tuple/setmap',
+        'tuple/slice', 'tuple/sourcemap', 'tuple/type',
+        'tuple?', 'type', 'unmarshal', 'untrace', 'update',
+        'update-in', 'values', 'varglobal', 'walk',
+        'warn-compile', 'xprin', 'xprinf', 'xprint', 'xprintf',
+        'yield', 'zero?', 'zipcoll',
+        # obsolete builtin functions
+        'tarray/buffer', 'tarray/copy-bytes', 'tarray/length',
+        'tarray/new', 'tarray/properties', 'tarray/slice',
+        'tarray/swap-bytes', 'thread/close', 'thread/current',
+        'thread/exit', 'thread/new', 'thread/receive',
+        'thread/send'
+    )
+
+    builtin_variables = (
+        'debugger-env', 'default-peg-grammar', 'janet/build',
+        'janet/config-bits', 'janet/version', 'load-image-dict',
+        'make-image-dict', 'math/-inf', 'math/e', 'math/inf',
+        'math/int-max', 'math/int-min', 'math/int32-max',
+        'math/int32-min', 'math/nan', 'math/pi', 'module/cache',
+        'module/loaders', 'module/loading', 'module/paths',
+        'root-env', 'stderr', 'stdin', 'stdout'
+    )
+
+    constants = (
+        'false', 'nil', 'true'
+    )
+
+    # XXX: this form not usable to pass to `suffix=`
+    #_token_end = r'''
+    #  (?=            # followed by one of:
+    #      \s           # whitespace
+    #    | \#           # comment
+    #    | [)\]]        # end delimiters
+    #    | $            # end of file
+    #  )
+    #'''
+
+    # ...so, express it like this
+    _token_end = r'(?=\s|#|[)\]]|$)'
+
+    _first_char = r'[a-zA-Z!$%&*+./<?=>@^_-]'
+    _following_chars_mb = r'[0-9:a-zA-Z!$%&*+./<?=>@^_-]*'
```

Comment:
I've removed all use of `_mb`.  It was supposed to stand for `may be`, but if it were to be left in, it would have been better to explain this somewhere.